CMake doesn't find packages properly

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 4.4.6-45-g3b1d59d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3b1d59d8c4fbf8c0ba0c9752007ff27bec3bc8b3
Last commit: 26 hours ago
Core tap JSON: 23 Nov 21:45 UTC
Core cask tap JSON: 23 Nov 21:45 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false
Your system is ready to brew.
```


### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Building a C++ project which depends on Eigen3 as well as on Ceres. It's reproducible on 2 different machines.

### What happened (include all command output)?

Running `cmake` on a project which finds Eigen3 before it finds Ceres results in an error.
```
CMake Error at /opt/homebrew/lib/cmake/Ceres/CeresConfig.cmake:89 (message):
  Failed to find Ceres - Found Eigen dependency, but the version of Eigen
  found () does not exactly match the version of Eigen Ceres was compiled
  with (3.4.0).  This can cause subtle bugs by triggering violations of the
  One Definition Rule.  See the Wikipedia article
  http://en.wikipedia.org/wiki/One_Definition_Rule for more details
Call Stack (most recent call first):
  /opt/homebrew/lib/cmake/Ceres/CeresConfig.cmake:204 (ceres_report_not_found)
  CMakeLists.txt:2 (find_Package)
```

### What did you expect to happen?

The same project used to build in previous Cmake versions.

Here is a minimal `CMakeLists.txt` which reproduces this error:
```
find_package(Eigen3)
find_package(Ceres)
```

This project works on the current upstream Cmake version, but fails with the above error on brew's Cmake. I've narrowed it down to the following patch:
d69d9821e263e4fcdcc114e44694f07e6524971e

Without this patch, the project builds and with it it doesn't. I suspect the reason is that it fails to find the `Eigen3Config.cmake` with the patch. Without the patch I can see in the `CMakeCache.txt` that the variable
```
//The directory containing a CMake configuration file for Eigen3.
Eigen3_DIR:PATH=/opt/homebrew/share/eigen3/cmake
```
is set. With the patch it is not set:
```
//The directory containing a CMake configuration file for Eigen3.
Eigen3_DIR:PATH=Eigen3_DIR-NOTFOUND
```

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
# This formula works on the minimal CMakeLists.txt:
brew uninstall cmake; rm ./cmake.rb; wget https://raw.githubusercontent.com/Homebrew/homebrew-core/c893863cdc5766f3aaea2a54cb6fae61ac9caea4/Formula/c/cmake.rb && brew install ./cmake.rb
rm -rf build; mkdir build && cd build && cmake ..; cd ..

# This formula with the patch doesn't work:
brew uninstall cmake; rm ./cmake.rb; wget https://raw.githubusercontent.com/Homebrew/homebrew-core/961a45133d237882bb410cdf2b72384798d2ae7a/Formula/c/cmake.rb && brew install ./cmake.rb
rm -rf build; mkdir build && cd build && cmake ..; cd ..
```

--- a/Formula/e/eigen.rb
+++ b/Formula/e/eigen.rb
@@ -24,7 +24,6 @@ class Eigen < Formula
   def install
     system "cmake", "-S", ".", "-B", "eigen-build", "-Dpkg_config_libdir=#{lib}", *std_cmake_args
     system "cmake", "--install", "eigen-build"
-    (share/"cmake/Modules").install "cmake/FindEigen3.cmake"
   end
 
   test do

--- a/Formula/e/eigen.rb
+++ b/Formula/e/eigen.rb
@@ -13,8 +13,8 @@ class Eigen < Formula
   end
 
   bottle do
-    rebuild 2
-    sha256 cellar: :any_skip_relocation, all: "b7d3fc4023e664e69392994530a88aa0b6f2a81067da6d64727cf983db2c1bd1"
+    rebuild 3
+    sha256 cellar: :any_skip_relocation, all: "06503290dc3c07a67b8f582046b0a7f0bd68c2cb2da1e5bc071710de5ba7f5ec"
   end
 
   depends_on "cmake" => :build

